src/services/productService.ts

```diff
@@ -0,0 +1,27 @@
+import env from '@/utils/env';
+const projectKey = env.VITE_CTP_PROJECT_KEY;
+const apiUrl = env.VITE_CTP_API_URL;
+import { Product } from '@/types/products';
+import { isProduct } from '@/utils/typeguards';
+
+export default async function getProductByKey(key: string): Promise<Product> {
+  const token = localStorage.getItem('access_token')
+    ? localStorage.getItem('access_token')
+    : localStorage.getItem('anonymous_id');
+  const productResponse = await fetch(
+    `${apiUrl}${projectKey}/products/key=${key}`,
+    {
+      method: 'GET',
+      headers: {
+        Authorization: 'Bearer ' + token,
+      },
+    }
+  );
+  const res: unknown = await productResponse.json();
```

Comment:
don't we need to check  the response status?
```suggestion
if (!productResponse.ok) {
    throw new Error(`Failed to fetch product: ${productResponse.status}`);
  }
  const res: unknown = await productResponse.json();
```

Reply:
I'm checking if the response is a Product object. If it's not, so that means that it failed and I throw an error in this case